pkg/registry/core/pod/etcd/eviction.go

```diff
@@ -66,41 +77,53 @@ func (r *EvictionREST) Create(ctx api.Context, obj runtime.Object) (runtime.Obje
 		return nil, err
 	}
 	pod := obj.(*api.Pod)
-	pdbs, err := r.getPodDisruptionBudgets(ctx, pod)
-	if err != nil {
-		return nil, err
-	}
-
-	if len(pdbs) > 1 {
-		return &unversioned.Status{
-			Status:  unversioned.StatusFailure,
-			Message: "This pod has more than one PodDisruptionBudget, which the eviction subresource does not support.",
-			Code:    500,
-		}, nil
-	} else if len(pdbs) == 1 {
-		pdb := pdbs[0]
-		// Try to verify-and-decrement
-
-		// If it was false already, or if it becomes false during the course of our retries,
-		// raise an error marked as a 429.
-		ok, err := r.checkAndDecrement(pod.Namespace, pod.Name, pdb)
+	var rtStatus *unversioned.Status
+	err = retry.RetryOnConflict(EvictionsRetry, func() error {
+		pdbs, err := r.getPodDisruptionBudgets(ctx, pod)
 		if err != nil {
-			return nil, err
+			return err
 		}
 
-		if !ok {
-			return &unversioned.Status{
-				Status: unversioned.StatusFailure,
-				// TODO(mml): Include some more details about why the eviction is disallowed.
-				// Ideally any such text is generated by the DisruptionController (offline).
-				Message: "Cannot evict pod as it would violate the pod's disruption budget.",
-				Code:    429,
-				// TODO(mml): Add a Retry-After header.  Once there are time-based
-				// budgets, we can sometimes compute a sensible suggested value.  But
-				// even without that, we can give a suggestion (10 minutes?) that
-				// prevents well-behaved clients from hammering us.
-			}, nil
+		if len(pdbs) > 1 {
+			rtStatus = &unversioned.Status{
+				Status:  unversioned.StatusFailure,
+				Message: "This pod has more than one PodDisruptionBudget, which the eviction subresource does not support.",
+				Code:    500,
+			}
+			return nil
+		} else if len(pdbs) == 1 {
+			pdb := pdbs[0]
+			// Try to verify-and-decrement
+
+			// If it was false already, or if it becomes false during the course of our retries,
+			// raise an error marked as a 429.
+			ok, err := r.checkAndDecrement(pod.Namespace, pod.Name, pdb)
+			if err != nil {
+				return err
+			}
+
+			if !ok {
+				rtStatus = &unversioned.Status{
+					Status: unversioned.StatusFailure,
+					// TODO(mml): Include some more details about why the eviction is disallowed.
+					// Ideally any such text is generated by the DisruptionController (offline).
+					Message: "Cannot evict pod as it would violate the pod's disruption budget.",
+					Code:    429,
+					// TODO(mml): Add a Retry-After header.  Once there are time-based
+					// budgets, we can sometimes compute a sensible suggested value.  But
+					// even without that, we can give a suggestion (10 minutes?) that
+					// prevents well-behaved clients from hammering us.
+				}
+			}
 		}
+		return nil
+	})
+	if err != nil {
```

Comment:
This error is not always going to be an api error, right? For example, if `RetryOnConflict` times out it will return a `ErrWaitTimeout`. Also `checkAndDecrement` returns non-api errors.